python-avd/pyavd/_eos_designs/structured_config/network_services/prefix_lists.py

```diff
@@ -20,55 +22,48 @@ class PrefixListsMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def prefix_lists(self: AvdStructuredConfigNetworkServicesProtocol) -> list | None:
+    @structured_config_contributor
+    def prefix_lists(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
         """
-        Return structured config for prefix_lists.
+        Set the structured config for prefix_lists.
 
         Covers EVPN services in VRF "default" and redistribution of connected to BGP
         """
         # Get prefix-lists from EVPN services in VRF "default" (if any)
-        prefix_lists = self._prefix_lists_vrf_default()
+        self._set_prefix_lists_vrf_default()
 
         # Add prefix-list for VRFs where MLAG iBGP peering should not be redistributed
         if mlag_prefixes := self._mlag_ibgp_peering_subnets_without_redistribution:
-            prefix_list = {"name": "PL-MLAG-PEER-VRFS", "sequence_numbers": []}
+            sequence_numbers = EosCliConfigGen.PrefixListsItem.SequenceNumbers()
             for index, mlag_prefix in enumerate(mlag_prefixes):
                 sequence = 10 * (index + 1)
-                prefix_list["sequence_numbers"].append({"sequence": sequence, "action": f"permit {mlag_prefix}"})
-
-            prefix_lists.append(prefix_list)
-
-        if prefix_lists:
-            return prefix_lists
+                sequence_numbers.append_new(sequence=sequence, action=f"permit {mlag_prefix}")
 
-        return None
+            self.structured_config.prefix_lists.append_new(name="PL-MLAG-PEER-VRFS", sequence_numbers=sequence_numbers)
 
-    def _prefix_lists_vrf_default(self: AvdStructuredConfigNetworkServicesProtocol) -> list:
-        """prefix_lists for EVPN services in VRF "default"."""
+    def _set_prefix_lists_vrf_default(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
+        """Set the prefix_lists for EVPN services in VRF "default"."""
         if not self._vrf_default_evpn:
-            return []
+            return
 
         subnets = self._vrf_default_ipv4_subnets
         static_routes = self._vrf_default_ipv4_static_routes["static_routes"]
         if not subnets and not static_routes:
-            return []
+            return
```

Comment:
Given that we don't need to return anything anymore, you don't need these two lines anymore - if there is no subnets and no static_routes, the next two if statement will just not do anything.

Reply:
Fixed!